peps/pep-0759.rst

```diff
@@ -0,0 +1,472 @@
+true if the main purpose of such an index is just to avoid file size
+limitations. The external index approach also imposes a tricky UX on consumers
+of projects on the external index, requiring them to understand how CLI
+options such as ``--external-index-url`` work, along with the security
+implications of such flags. It would be much easier for both producers and
+consumers of large wheel packages to just set up and maintain a simple web
+server, capable of serving individual files with no more complex API than
+``HTTP GET``. Such an interface is also easily cacheable or placed behind a
+`CDN <https://en.wikipedia.org/wiki/Content_delivery_network>`__. Simple HTTP
+servers are also much easier to audit for security purposes, easier to proxy,
+and usually take much less resources to run, support, and maintain.  Even
+something like `Amazon S3 <https://aws.amazon.com/s3/>`__ could be used to
+host external wheels.
+
+This PEP proposes an approach that favors such operational simplicity.
+
+Specification
+=============
+
+A new type of uploadable file is defined, called a "RIM" (i.e. ``.rim``) file, so named to
+evoke the image of a wheel with the tire removed, and backronym'd to "Remote Installable
+Metadata" files. ``.rim`` files are easily derivable from wheels, and the process of
+turning a wheel into a ``.rim`` is outlined below. The file name format exactly matches
+the `wheel file naming format`_ specification, except that RIM files use the suffix
+``.rim``. This means that all the tags used to discriminate ``.whl`` files also
+distinguish between different ``.rim`` files, and thus can be used during dependency
+resolution steps, exactly as ``.whl`` files are today. In this respect, ``.whl`` and
+``.rim`` files are interchangeable.
+
+The content of a ``.rim`` file is *nearly* identical to ``.whl`` files, however ``.rim``
+files **MUST** contain only the ``.dist-info`` directory from a wheel. No other top-level
+file or directory is allowed in the ``.rim`` zip file. The ``.dist-info`` directory
+**MUST** contain a single additional file in addition to those `allowed`_ in a ``.whl``
+file's ``.dist-info`` directory: a file called ``EXTERNAL-HOSTING.json``.
+
+.. _file-format:
+
+This is a JSON file contains containing the following keys:
+
+``version``
+    This is the file format version, which for this PEP **MUST** be ``1.0``.
+``owner``
+    This **MUST** name the PyPI organization owner of this externally hosted file, for
+    reasons which will be described in detail below.
+``uri``
+    This is a single URL naming the location of the physical ``.whl`` file hosted on an
+    external site. This URL **MUST** use the ``https`` scheme.
+``size``
+    This is an integer value describing the size in bytes of the physical ``.whl`` file.
+``hashes``
+    This is a dictionary of the format described in :pep:`694`, used to capture both the
+    `hash algorithm and hex encoded digests`_ of the physical ``.whl`` file, with the same
```

Comment:
```suggestion
    :pep:`694#upload-each-file` of the physical ``.whl`` file, with the same
```

I *think* this is correct based on my understanding of the docs: https://www.sphinx-doc.org/en/master/usage/restructuredtext/roles.html#role-pep